tests/common.go

```diff
@@ -2912,6 +2914,198 @@ func DeleteBucket(provider string, bucketName string) {
+			Inst().V.String(), ctx.App.Key, v),
+		func() {
+			currRep, err := Inst().V.GetReplicationFactor(v)
+			if err != nil {
+				err = fmt.Errorf("error getting replication factor for volume %s, Error: %v", v.Name, err)
+				logrus.Error(err)
+				UpdateOutcome(event, err)
+				return
+			}
+
+			//if repl is 3 cannot increase repl for the volume
+			if currRep == 3 {
+				err = fmt.Errorf("cannot perform repl incease as current repl factor is %d", currRep)
+				logrus.Error(err)
+				UpdateOutcome(event, err)
+				return
+			}
+
+			if err == nil {
+				Step(
+					fmt.Sprintf("repl increase volume driver %s on app %s's volume: %v",
+						Inst().V.String(), ctx.App.Key, v),
+					func() {
+						replicaSets, err := Inst().V.GetReplicaSets(v)
+						if err == nil {
+							replicaNodes := replicaSets[0].Nodes
+							err = Inst().V.SetReplicationFactor(v, currRep+1, nil, false)
+							if err != nil {
+								logrus.Errorf("There is an error increasing repl [%v]", err.Error())
+								UpdateOutcome(event, err)
+							} else {
+								logrus.Info("Waiting for 10 seconds for re-sync to initialize before source nodes reboot")
+								time.Sleep(10 * time.Second)
+								//rebooting source nodes one by one
+								for _, nID := range replicaNodes {
+									replNodeToReboot := storageNodeMap[nID]
+									err = Inst().N.RebootNode(replNodeToReboot, node.RebootNodeOpts{
+										Force: true,
+										ConnectionOpts: node.ConnectionOpts{
+											Timeout:         1 * time.Minute,
+											TimeBeforeRetry: 5 * time.Second,
+										},
+									})
+									if err != nil {
+										logrus.Errorf("error rebooting node %v, Error: %v", replNodeToReboot.Name, err)
+										UpdateOutcome(event, err)
+									}
+								}
+								err = validateReplFactorUpdate(v, currRep+1)
+								if err != nil {
+									err = fmt.Errorf("error in ha-increse after  source node reboot. Error: %v", err)
+									logrus.Error(err)
+									UpdateOutcome(event, err)
+								} else {
+									logrus.Infof("repl successfully increased to %d", currRep+1)
+								}
+							}
+						} else {
+							err = fmt.Errorf("error getting relicasets for volume %s, Error: %v", v.Name, err)
+							logrus.Error(err)
+							UpdateOutcome(event, err)
+						}
+
+					})
+			} else {
+				err = fmt.Errorf("error getting current replication factor for volume %s, Error: %v", v.Name, err)
+				logrus.Error(err)
+				UpdateOutcome(event, err)
+			}
+
+		})
+}
+
+func validateReplFactorUpdate(v *volume.Volume, expaectedReplFactor int64) error {
+	t := func() (interface{}, bool, error) {
+		err := Inst().V.WaitForReplicationToComplete(v, expaectedReplFactor, validateReplicationUpdateTimeout)
+		if err != nil {
+			statusErr, _ := status.FromError(err)
+			if statusErr.Code() == codes.NotFound || strings.Contains(err.Error(), "code = NotFound") {
+				return nil, false, err
```

Comment:
here also returning 3 values ?

Reply:
yes, this code block is part of DoRetryWithTimeout where it expects (out, retry, err).
If err it is returining no retry and err